consumer.go

```diff
@@ -376,11 +393,13 @@ func (child *PartitionConsumer) Close() error {
 	// close itself)
 	close(child.dying)
 
+	for _ = range child.messages {
```

Comment:
If the user calls close, then the internal goroutines produce a flood of errors, this will hang waiting for success channel to close, and internal will hang trying to write errors